croncat/src/store/agent.rs

```diff
@@ -183,10 +199,23 @@ impl LocalAgentStorage {
         let mnemonic: Mnemonic = entry.mnemonic.parse()?;
         let key = cosmrs::bip32::XPrv::derive_from_path(
             mnemonic.to_seed(""),
-            &utils::DERVIATION_PATH.parse().unwrap(),
+            &utils::DERIVATION_PATH.parse().unwrap(),
         )?;
         Ok(key)
     }
+
+    /// Retrieve an agent based on the key
+    fn get(&self, account_id: &str) -> Option<&LocalAgentStorageEntry> {
+        info!("Getting agent by id: {}", account_id);
+
+        let found = self.data.get(account_id);
+
+        if found.is_some() {
+            info!("Found agent: {:#?}", found.unwrap());
```

Comment:
Sorry for being pendantic here, I'm pretty sure it's optimized, so it doesn't look twice inside `Option` here, but anyway:
```suggestion
        if let Some(entry) = found {
            info!("Found agent: {:#?}", entry);
```

Reply:
Not too pendantic at all! Much better choice of syntax.